### PR TITLE
fix rebuild issue

### PR DESCRIPTION
**Pull request by blkart**
* Fix rebuild of an instance with a volume attached
* Replace root volume during rebuild
* rebuild: make sure server is shut down before volumes are detached

redmine #6426

### CODE REVIEW

File: nova/compute/manager.py [hunk 1]
```diff
@@ -2649,6 +2649,44 @@ def restore_instance(self, context, instance):
...
+                                         metadata=metadata,
+                                         availability_zone=availability_zone,
+                                         image_id=image_meta['id'])
+        self._await_block_device_map_created(context, new_vol['id'])
+        LOG.debug('Created new root volume %s. Updating bdm.',
+                  new_vol['id'])
+        bdm.volume_id = new_vol['id']
+        bdm.connection_info = None
+        bdm.image_id = image_meta['id']
+        return bdm
```
Comment (plysan) [inline nova/compute/manager.py:2689]: 下面空一行

File: nova/compute/manager.py [hunk 2]
```diff
@@ -2665,6 +2703,23 @@ def _rebuild_default_impl(self, context, instance, image_meta,
...
+                        self.volume_api.delete(context, old_vol_id)
+                    except Exception as exc:
+                        LOG.warning(_LW('Failed to delete volume: '
+                                        '%(volume_id)s due to %(exc)s'),
+                                    {'volume_id': root_bdm.volume_id,
+                                     'exc': exc})
+                bdms = list(block_device.get_bdms_to_connect(
+                    bdms, exclude_root_mapping=True))
+                bdms.insert(0, new_root_bdm)
+
```
Comment (plysan) [inline nova/compute/manager.py]: 上面空行应该删掉

**Review by plysan — COMMENTED** [review]
* 后两个 commit 信息超过行宽没有换行

File: nova/compute/manager.py (hunk 1 above)
Comment (blkart) [reply nova/compute/manager.py:2689]: 已添加空行

**Pull request merged**